azext_edge/edge/providers/check/base/resource.py

```diff
@@ -468,3 +470,19 @@ def process_custom_resource_status(
                     namespace=namespace,
                     display=Padding(status_text, (0, 0, 0, padding + 4)),
                 )
+
+
+def validate_ref(name: str, namespace: str, ref_type: ValidationResourceType) -> Tuple[str, bool]:
+    ref_obj = None
+    text = ""
+    if ref_type == ValidationResourceType.secret:
+        ref_obj = get_namespaced_secret(secret_name=name, namespace=namespace)
+    elif ref_type == ValidationResourceType.configmap:
+        ref_obj = get_config_map(name=name, namespace=namespace)
+    is_valid = bool(ref_obj)
+
+    if is_valid:
+        text = f"[green]Valid[/green] {ref_type.value} reference {{[green]{name}[/green]}}."
+    else:
+        text = f"[red]Invalid[/red] {ref_type.value} reference {{[red]{name}[/red]}}."
+    return text, is_valid
```

Comment:
got it, in that case tho should I just call the get_namespaced_{resource} from the individual service check file? I don't see an example of that in our code, I only see those called in base files, so that's another reason I added a wrapper here